light-clients/ics10-grandpa-cw/src/contract.rs

```diff
@@ -269,16 +269,27 @@ fn process_message(
 
 			// Check that the substitute client state is valid:
 			// all fields should be the same as in the old state, except for the `relay_chain`,
-			// `para_id`, `latest_para_height`, `latest_relay_height`, `frozen_height`,
-			// `current_authorities`, `current_set_id`
-			old_client_state.relay_chain = substitute_client_state.relay_chain;
-			old_client_state.para_id = substitute_client_state.para_id;
-			old_client_state.latest_para_height = substitute_client_state.latest_para_height;
-			old_client_state.latest_relay_height = substitute_client_state.latest_relay_height;
-			old_client_state.frozen_height = substitute_client_state.frozen_height;
-			old_client_state.current_authorities =
-				substitute_client_state.current_authorities.clone();
-			old_client_state.current_set_id = substitute_client_state.current_set_id;
+			// `para_id`, `latest_para_height`, `latest_relay_height`, `latest_relay_hash`,
+			// `frozen_height`, `current_authorities`, `current_set_id`
+			let ClientState {
+				relay_chain,
+				latest_relay_height,
+				latest_relay_hash,
+				frozen_height,
+				latest_para_height,
+				para_id,
+				current_set_id,
+				current_authorities,
+				_phantom,
+			} = substitute_client_state.clone();
+			old_client_state.relay_chain = relay_chain;
```

Comment:
Can we just add a function that does it? And in some way ensure that we are exhaustive. Perhaps just building an entire new one?

Reply:
This is a good way to make sure the we're exhaustive, because if we add new fields, the compilation will fail, because the fields are not mentioned here